presto-main/src/main/java/com/facebook/presto/operator/DynamicFilterSourceOperator.java

```diff
@@ -0,0 +1,266 @@
+        {
+            this.filterId = filterId;
+            this.type = type;
+            this.index = index;
+        }
+    }
+
+    public static class DynamicFilterSourceOperatorFactory
+            implements OperatorFactory
+    {
+        private final int operatorId;
+        private final PlanNodeId planNodeId;
+        private final Consumer<TupleDomain<String>> dynamicPredicateConsumer;
+        private final List<Channel> channels;
+        private final int maxFilterPositionsCount;
+        private final DataSize maxFilterSize;
+
+        private boolean closed;
+
+        public DynamicFilterSourceOperatorFactory(
+                int operatorId,
+                PlanNodeId planNodeId,
+                Consumer<TupleDomain<String>> dynamicPredicateConsumer,
+                List<Channel> channels,
+                int maxFilterPositionsCount,
+                DataSize maxFilterSize)
+        {
+            this.operatorId = operatorId;
+            this.planNodeId = requireNonNull(planNodeId, "planNodeId is null");
+            this.dynamicPredicateConsumer = requireNonNull(dynamicPredicateConsumer, "dynamicPredicateConsumer is null");
+            this.channels = requireNonNull(channels, "channels is null");
+            verify(channels.stream().map(channel -> channel.filterId).collect(toSet()).size() == channels.size(),
+                    "duplicate dynamic filters are not allowed");
+            verify(channels.stream().map(channel -> channel.index).collect(toSet()).size() == channels.size(),
+                    "duplicate channel indices are not allowed");
+            this.maxFilterPositionsCount = maxFilterPositionsCount;
+            this.maxFilterSize = maxFilterSize;
+        }
+
+        @Override
+        public Operator createOperator(DriverContext driverContext)
+        {
+            checkState(!closed, "Factory is already closed");
+            return new DynamicFilterSourceOperator(
+                    driverContext.addOperatorContext(operatorId, planNodeId, DynamicFilterSourceOperator.class.getSimpleName()),
+                    dynamicPredicateConsumer,
+                    channels,
+                    planNodeId,
+                    maxFilterPositionsCount,
+                    maxFilterSize);
+        }
+
+        @Override
+        public void noMoreOperators()
+        {
+            checkState(!closed, "Factory is already closed");
+            closed = true;
+        }
+
+        @Override
+        public OperatorFactory duplicate()
+        {
+            throw new UnsupportedOperationException("duplicate() is not supported for DynamicFilterSourceOperatorFactory");
+        }
+    }
+
+    private final OperatorContext context;
+    private boolean finished;
+    private Page current;
+    private final Consumer<TupleDomain<String>> dynamicPredicateConsumer;
+    private final int maxFilterPositionsCount;
+    private final long maxFilterSizeInBytes;
+
+    private final List<Channel> channels;
```

Comment:
Let's move all final variables together and non-final together. Break a line between these two.